pkg/operator/controller/ingress/status_test.go

```diff
@@ -1127,3 +1128,87 @@ func TestMergeConditions(t *testing.T) {
 		}
 	}
 }
+
+func TestZoneInConfig(t *testing.T) {
+	var z *configv1.DNSZone
+	var dnsZone configv1.DNSZone
+	tag := make(map[string]string)
+	tagZone := make(map[string]string)
+
+	testCases := []struct {
+		description        string
+		expected           bool
+		in, zone, zoneType string
+	}{
+		{
+			description: "[PrivateZone] empty strings (should fail)",
+			expected:    false,
+			in:          "",
+			zone:        "",
+			zoneType:    "ID",
+		},
+		{
+			description: "[PrivateZone] zone.ID empty string (should fail)",
+			expected:    false,
+			in:          "test",
+			zone:        "",
+			zoneType:    "ID",
+		}, {
+			description: "[PrivateZone] zone.ID with value (not equal should fail)",
+			expected:    false,
+			in:          "test",
+			zone:        "notest",
+			zoneType:    "ID",
+		}, {
+			description: "[PrivateZone] zone.ID with value (equal should pass)",
+			expected:    true,
+			in:          "test",
+			zone:        "test",
+			zoneType:    "ID",
+		}, {
+			description: "[PrivateZone] empty strings (should fail)",
+			expected:    false,
+			in:          "",
+			zone:        "",
+			zoneType:    "TAG",
+		}, {
+			description: "[PrivateZone] zone.Tags['Name'] empty string (should fail)",
+			expected:    false,
+			in:          "test",
+			zone:        "",
+			zoneType:    "TAG",
+		}, {
+			description: "[PrivateZone] zone.Tags['Name'] with value (not equal should fail)",
+			expected:    false,
+			in:          "test",
+			zone:        "notest",
+			zoneType:    "TAG",
+		}, {
+			description: "[PrivateZone] zone.tags['Name'] with value (equal should pass)",
+			expected:    true,
+			in:          "test",
+			zone:        "test",
+			zoneType:    "TAG",
+		},
+	}
+
+	for _, test := range testCases {
+		t.Logf("test: %s", test.description)
+		if test.zoneType == "ID" {
+			z = &configv1.DNSZone{ID: test.in}
+			dnsZone = configv1.DNSZone{ID: test.zone}
+		} else {
+			tag["Name"] = test.in
+			z = &configv1.DNSZone{Tags: tag}
+			tagZone["Name"] = test.zone
+			dnsZone = configv1.DNSZone{Tags: tagZone}
+		}
+		dnsSpec := configv1.DNSSpec{PrivateZone: z}
+		dnsConfig := &configv1.DNS{Spec: dnsSpec}
+		actual := checkZoneInConfig(dnsConfig, dnsZone)
+		if actual != test.expected {
+			t.Errorf("expected:%v actual:%v\n", test.expected, actual)
+		}
+	}
```

Comment:
Consider using `t.Run()`:

```suggestion
	for _, test := range testCases {
		t.Run(test.description, func(t *testing.T) {
			if test.zoneType == "ID" {
				z = &configv1.DNSZone{ID: test.in}
				dnsZone = configv1.DNSZone{ID: test.zone}
			} else {
				tag["Name"] = test.in
				z = &configv1.DNSZone{Tags: tag}
				tagZone["Name"] = test.zone
				dnsZone = configv1.DNSZone{Tags: tagZone}
			}
			dnsSpec := configv1.DNSSpec{PrivateZone: z}
			dnsConfig := &configv1.DNS{Spec: dnsSpec}
			actual := checkZoneInConfig(dnsConfig, dnsZone)
			if actual != test.expected {
				t.Errorf("expected:%v actual:%v\n", test.expected, actual)
			}
		})
	}
```